src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -159,41 +165,55 @@ public void handle_package_result(PackageResult packageResult, ChocolateyConfigu
                 pkgInfo.IsSideBySide = true;
             }
 
-            if (packageResult.Success && config.Information.PlatformType == PlatformType.Windows)
+            if (packageResult.Success)
             {
                 if (!config.SkipPackageInstallProvider)
                 {
-                    var before = _registryService.get_installer_keys();
+                    Registry before = null;
+                    if (config.Information.PlatformType == PlatformType.Windows)
+                    {
+                        before = _registryService.get_installer_keys();
+                    }
 
-                    var powerShellRan = _powershellService.install(config, packageResult);
-                    if (powerShellRan)
+                    var providerRan = false;
+                    foreach (var installProvider in _installProviders.Where(ip => ip.can_be_used(config)).or_empty_list_if_null())
                     {
-                        //todo: prevent reboots
+                        if (providerRan == true) break;

```

Comment:
Short-circuit. :+1: 

We also need a way for users to set a preference on which they would like to see run first.